### PR TITLE
o/snapstate, features: add feature flags for disk space awareness

### PR DESCRIPTION
**Pull request by stolowski**
Add feature flags for disk space awareness (separately for install, remove and refresh). Only install/remove are actually used at the moment, refresh check will come in a followup.

### CODE REVIEW

**Comment by pedronis** [comment]
doesn't this need to be split?  we need to backport the remove bit to 2.46

cc @mvo5

**Comment by stolowski** [comment]
> doesn't this need to be split? we need to backport the remove bit to 2.46
> 
> cc @mvo5

Of course, you're right! I'll split it.

**Comment by mvo5** [comment]
Thank you! Fwiw, this looks very nice. But Samuele is right, we want a commit that we can cherry pick for 2.46 :)

**Comment by stolowski** [comment]
> Thank you! Fwiw, this looks very nice. But Samuele is right, we want a commit that we can cherry pick for 2.46 :)

Yes, done in #9251

**Comment by stolowski** [comment]
The feature flag for remove was landed separately and this PR now adds the remaining flags.

File: overlord/snapstate/snapstate_test.go [hunk 1]
```diff
@@ -1100,7 +1100,7 @@ func (s *snapmgrTestSuite) TestRemoveDiskSpaceForSnapshotError(c *C) {
 	c.Check(diskSpaceErr.ChangeKind, Equals, "remove")
 }
 
-func (s *snapmgrTestSuite) TestRemoveDiskSpaceForSnapshotNotCheckedWhenSnapshotsDisabled(c *C) {
+func (s *snapmgrTestSuite) TestRemoveDiskSpaceCheckDisabled(c *C) {
```
Comment (pedronis) [inline overlord/snapstate/snapstate_test.go:1103]: what's happening here, did you switch the two tests around? the diff is a bit confusing

Comment (stolowski) [reply overlord/snapstate/snapstate_test.go:1103]: Yes, that happended when I merged master (with already landed remove bit) back and deconflicted it.

File: overlord/snapstate/snapstate_test.go [hunk 2]
```diff
@@ -1122,20 +1126,15 @@ func (s *snapmgrTestSuite) TestRemoveDiskSpaceForSnapshotNotCheckedWhenSnapshots
 	})
 
 	_, err := snapstate.Remove(s.state, "some-snap", snap.R(0), nil)
-	c.Assert(err, IsNil)
+	c.Check(err, IsNil)
 	c.Assert(automaticSnapshotCalled, Equals, true)
 }
 
-func (s *snapmgrTestSuite) TestRemoveDiskSpaceCheckDisabled(c *C) {
+func (s *snapmgrTestSuite) TestRemoveDiskSpaceForSnapshotNotCheckedWhenSnapshotsDisabled(c *C) {
```
Comment (pedronis) [inline overlord/snapstate/snapstate_test.go:1133]: this and TestRemoveDiskSpaceCheckDisabled seems to be the same test ? what am I missing?

Comment (pedronis) [reply overlord/snapstate/snapstate_test.go:1133]: ah, one disables the feature explicitly and one uses the default?

Comment (stolowski) [reply overlord/snapstate/snapstate_test.go:1133]: I've refactored these tests separately in #9274

File: overlord/snapstate/snapstate_test.go (hunk 1 above)
Comment (stolowski) [reply overlord/snapstate/snapstate_test.go]: That's no longer an issue in this PR.

**Review by pedronis — APPROVED** [review]
thank you

**Review by mvo5 — APPROVED** [review]
Thank you!

File: features/features_test.go [hunk 3]
```diff
@@ -50,6 +50,8 @@ func (*featureSuite) TestName(c *C) {
 	c.Check(features.UserDaemons.String(), Equals, "user-daemons")
 	c.Check(features.DbusActivation.String(), Equals, "dbus-activation")
 	c.Check(features.HiddenSnapFolder.String(), Equals, "hidden-snap-folder")
+	c.Check(features.CheckDiskSpaceInstall.String(), Equals, "check-disk-space-install")
```
Comment (zyga) [inline features/features_test.go:53]: They are not in the same order as above.

**Review by zyga — APPROVED** [review]
LGTM, one commet for follow-up

**Pull request merged**